src/main/kotlin/com/atlassian/performance/tools/virtualusers/LoadTest.kt

```diff
@@ -110,29 +106,16 @@ internal class LoadTest(
             LinkedBlockingQueue<Runnable>(),
             ThreadFactoryBuilder().setNameFormat("virtual-user-%d").setDaemon(true).build()
         )
-        val stopSchedule = Executors.newScheduledThreadPool(
-            1,
-            ThreadFactoryBuilder().setNameFormat("deferred-stop").setDaemon(true).build()
-        )
         val segments = (1..virtualUsers).map { segmentLoad(it) }
-        val stop = stopSchedule.schedule(
-            {
-                val active = loadPool.activeCount
-                logger.info("Stopping load")
-                loadPool.shutdownNow()
-                segments.forEach { it.close() }
-                if (active != virtualUsers) {
-                    throw Exception("Expected $virtualUsers VUs to still be active, but encountered $active")
-                }
-            },
-            finish.toMillis(),
-            TimeUnit.MILLISECONDS
-        )
-        val deadline = now() + finish + maxStop
-        logger.info("Deadline for tests is $deadline.")
         segments.forEach { loadPool.submit { applyLoad(it) } }
-        stop.finishBy(deadline, logger)
-        stopSchedule.shutdownNow()
+        Thread.sleep(finish.toMillis())
+        logger.info("Stopping load")
+        val active = loadPool.activeCount
+        segments.forEach { it.close() }
+        logger.info("Segments closed")
```

Comment:
https://gist.github.com/dagguh/b663b34d74d15438b7c7d959c092c78e :
```
2019-04-24T10:06:00,163Z INFO  {} [] [LoadTest] Stopping load
...
2019-04-24T10:13:52,763Z ERROR {} [applying load #74e68804-6bf2-44d6-abc9-79a2c0065316] [ExploratoryVirtualUser] Done applying load
...
2019-04-24T10:13:52,848Z ERROR {} [applying load #00728680-b029-4c0c-896a-c3b3d84dbc90] [ExploratoryVirtualUser] Done applying load
...
2019-04-24T10:14:35,467Z INFO  {} [] [LoadTest] Segments closed
```

Also there's 12 counts of `Done applying load`, 1 for each VU. They all are preceded with an error like:
```
2019-04-24T10:13:52,848Z ERROR {} [applying load #00728680-b029-4c0c-896a-c3b3d84dbc90] [ExploratoryVirtualUser] Failed to run com.atlassian.performance.tools.jiraactions.api.action.SearchJqlAction@34b80094, but we keep running
java.lang.Exception: Failed to run com.atlassian.performance.tools.jiraactions.api.action.SearchJqlAction@34b80094
	at com.atlassian.performance.tools.virtualusers.ExploratoryVirtualUser.runWithDiagnostics(ExploratoryVirtualUser.kt:82) ~[virtual-users.jar:?]
	at com.atlassian.performance.tools.virtualusers.ExploratoryVirtualUser.applyLoad(ExploratoryVirtualUser.kt:56) [virtual-users.jar:?]
	at com.atlassian.performance.tools.virtualusers.LoadTest.applyLoad(LoadTest.kt:169) [virtual-users.jar:?]
	at com.atlassian.performance.tools.virtualusers.LoadTest.access$applyLoad(LoadTest.kt:35) [virtual-users.jar:?]
	at com.atlassian.performance.tools.virtualusers.LoadTest$applyLoad$$inlined$forEach$lambda$1.run(LoadTest.kt:110) [virtual-users.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_191]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_191]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_191]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_191]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_191]
Caused by: java.io.IOException: Stream closed
	at java.io.BufferedWriter.ensureOpen(BufferedWriter.java:116) ~[?:1.8.0_191]
	at java.io.BufferedWriter.write(BufferedWriter.java:221) ~[?:1.8.0_191]
	at java.io.Writer.write(Writer.java:157) ~[?:1.8.0_191]
	at java.io.Writer.append(Writer.java:227) ~[?:1.8.0_191]
	at java.io.Writer.append(Writer.java:50) ~[?:1.8.0_191]
	at com.atlassian.performance.tools.jiraactions.api.measure.output.AppendableActionMetricOutput.write(AppendableActionMetricOutput.kt:15) ~[virtual-users.jar:?]
	at com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter.record(ActionMeter.kt:122) ~[virtual-users.jar:?]
	at com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter.measure(ActionMeter.kt:79) ~[virtual-users.jar:?]
	at com.atlassian.performance.tools.jiraactions.api.action.SearchJqlAction.run(SearchJqlAction.kt:21) ~[virtual-users.jar:?]
	at com.atlassian.performance.tools.virtualusers.ExploratoryVirtualUser.runWithDiagnostics(ExploratoryVirtualUser.kt:79) ~[virtual-users.jar:?]
	... 9 more
```
The error is expected. However, I didn't expect we'd be de facto waiting for all VU threads. Does `LoadSegment.close` block until the VU thread is done with it?

Reply:
`LoadTestIT#shouldTerminateDespiteSlowUninterruptibleNavigation` somewhat reproduces the blocking `close` problem.
As is, you see this:
```
2019-04-24T12:24:48,943Z INFO  main [] [com.atlassian.performance.tools.virtualusers.LoadTest] Stopping load
...
2019-04-24T12:24:48,957Z INFO  virtual-user-0 [applying load #02e4476e-fbd2-4694-92f0-6e235642c158] [com.atlassian.performance.tools.virtualusers.ExploratoryVirtualUser] Done applying load
2019-04-24T12:24:50,979Z INFO  main [] [com.atlassian.performance.tools.virtualusers.LoadTest] Segments closed
```

If you bump `server.stop(2)` to `server.stop(15)` you see this:
```
2019-04-24T12:25:55,614Z INFO  main [] [com.atlassian.performance.tools.virtualusers.LoadTest] Stopping load
...
2019-04-24T12:25:55,629Z INFO  virtual-user-0 [applying load #c6a58586-5866-48e3-9963-0910f422f9a5] [com.atlassian.performance.tools.virtualusers.ExploratoryVirtualUser] Done applying load
2019-04-24T12:26:10,774Z INFO  main [] [com.atlassian.performance.tools.virtualusers.LoadTest] Segments closed
```

Reply:
The repro is merged in and fixed